src/api/docgen/view/documentation.ts

```diff
@@ -0,0 +1,114 @@
+import * as reflect from "jsii-reflect";
+import { Markdown } from "../render/markdown";
+import { PythonTranspile } from "../transpile/python";
+import { Transpile } from "../transpile/transpile";
+import { ApiReference } from "./api-reference";
+import { Readme } from "./readme";
+
+/**
+ * Options for rendering documentation pages.
+ */
+export interface DocumentationOptions {
+  /**
+   * The assembly to generate from.
+   */
+  readonly assembly: reflect.Assembly;
+
+  /**
+   * Which language to generate docs for.
+   */
+  readonly language: string;
+
+  /**
+   * Include a generated api reference in the documentation.
+   *
+   * @default true
+   */
+  readonly apiReference?: boolean;
+
+  /**
+   * Include the user defined README.md in the documentation.
+   *
+   * @default true
+   */
+  readonly readme?: boolean;
+
+  /**
+   * Generate documentation only for a specific submodule.
+   *
+   * @default - Documentation is generated for all submodules.
+   */
+  readonly submoduleName?: string;
+}
+
+/**
+ * Render documentation pages for a jsii library.
+ */
+export class Documentation {
+  private readonly assembly: reflect.Assembly;
+  private readonly submodule?: reflect.Submodule;
+
+  private readonly includeReadme: boolean;
+  private readonly includeApiReference: boolean;
+
+  private readonly transpile: Transpile;
+
+  constructor(options: DocumentationOptions) {
+    this.assembly = options.assembly;
+
+    switch (options.language) {
+      case "python":
+        this.transpile = new PythonTranspile(this.assembly.system);
+        break;
+      default:
+        throw new Error(
+          `Generating documentation for ${options.language} is not supported`
+        );
+    }
+
+    if (options.submoduleName) {
+      this.submodule = this.findSubmodule(options.submoduleName);
+    }
+
+    this.includeApiReference = options.apiReference ?? true;
+    this.includeReadme = options.readme ?? false;
+  }
+
+  /**
+   * Generate markdown.
+   */
+  public get markdown(): Markdown {
+    const documentation = new Markdown();
+
+    if (this.includeReadme) {
+      const readme = new Readme(this.transpile, this.assembly, this.submodule);
+      documentation.section(readme.markdown);
+    }
+    if (this.includeApiReference) {
+      const apiReference = new ApiReference(
+        this.transpile,
+        this.assembly,
+        this.submodule
+      );
+      documentation.section(apiReference.markdown);
+    }
+
+    return documentation;
+  }
+
+  private findSubmodule(submoduleName: string): reflect.Submodule {
+    const submodules = this.assembly.submodules.filter(
+      (s) => s.name === submoduleName
+    );
+
+    if (submodules.length === 0) {
+      throw new Error(`Submodule ${submoduleName} not found`);
```

Comment:
An exception is more appropriate because there is nothing we can do if the submodule doesn't exist, this component shouldn't exist and shouldn't be initialized. 

Maybe we should introduce a specific exception type so that callers can easily identify such situations, but this can be added easily when such a use-case arises (probably soon with proper UI error handling).

Reply:
I wouldn't postpone the merge of this PR for that, but I do think informative error handling should be added - whether by informative response code or by informative exceptions, depending on the scenario